packages/password/src/index.mts

```diff
@@ -19,6 +23,7 @@ const password: Prompt<string, PasswordConfig> = (config, context) => {
     {
       ...config, // Make sure we do not display the default password
       default: undefined,
+      allowSkipTransformer: config.allowShowPassword,
```

Comment:
I'm not sure this is the right approach.

Approaching the implementation this way, we're limiting ourselves to having the implementation lives inside the `input` prompt. One thing for example that's missing now is an indication of the shortcut to unmask the password.

I feel like this feature addition might call for implementation a password prompt that is not a passthrough to the input one. At the cost of duplicating some code (but I don't think it's so much to be problematic 🤔.)

The alternative would be to work on actual prompt composition within the core (right now it wouldn't work out of the box.) Thinking about this option quickly, it feels like a more complex code change and I'm not sure where else we would want that capacity. So unclear to me if it's worth the effort VS some code duplication.

Reply:
Yeah, I do agree that having the implementation be only in the input prompt is a bit limiting.

So, I do think that the idea of implementing a password prompt might be the right call. While we will have some code duplication, we will have more freedom with managing the two prompts independently, and I do think in the end it looks cleaner (for example, we will not anymore to pass `default: undefined` to avoid having a default value)

Prompt composition is a bit overkill for this task and I do not think it's worth the effort...

I will then revert the changes made and proceed with the implementation of a password prompt